From c206cc2cd695ae0da93332dfbc6fc084de045b48 Mon Sep 17 00:00:00 2001
From: Tim Yao <tim.yao@amlogic.com>
Date: Tue, 27 Jan 2015 21:11:42 -0800
Subject: [PATCH 5380/5965] PD #88700-3. H.264 4K decoder: Add time delay
 before releasing CMA buffer to make sure the last saved buffer is the final
 displayed buffer.

Change-Id: I234e5a1b232b7ba91d970dfa25429314d35f3140
---
 drivers/amlogic/amports/vh264_4k2k.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index 763246a4b33c..f18d1f1846a6 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -1372,6 +1372,10 @@ static int vh264_4k2k_stop(void)
         amvdec2_disable();
     }
 
+#ifdef CONFIG_VSYNC_RDMA
+    msleep(100);
+#endif
+
     canvas_read((READ_VCBUS_REG(VD1_IF0_CANVAS0) & 0xff), &cur_canvas);
     disp_addr = cur_canvas.addr;
 
-- 
2.19.0

